src/QMCApp/qmcapp.cpp

```diff
@@ -43,205 +44,220 @@ void output_hardware_info(Communicate* comm, Libxml2Document& doc, xmlNodePtr ro
+    {
+      qmcComm               = new Communicate(*OHMMS::Controller, inputs.size());
+      qmc_common.mpi_groups = inputs.size();
+    }
+    std::stringstream logname;
+    int inpnum          = (inputs.size() > 1) ? qmcComm->getGroupID() : 0;
+    std::string myinput = inputs[qmcComm->getGroupID()];
+    myinput             = myinput.substr(0, myinput.size() - 4);
+    logname << myinput;
+
+    if (qmcComm->rank() != 0)
+    {
+      outputManager.shutOff();
+      // might need to redirect debug stream to a file per rank if debugging is enabled
+    }
+    if (inputs.size() > 1 && qmcComm->rank() == 0)
+    {
+      char fn[128];
+      snprintf(fn, 127, "%s.g%03d.qmc", logname.str().c_str(), qmcComm->getGroupID());
+      fn[127] = '\0';
+      infoSummary.redirectToFile(fn);
+      infoLog.redirectToSameStream(infoSummary);
+      infoError.redirectToSameStream(infoSummary);
+    }
+
+    //#if defined(MPIRUN_EXTRA_ARGUMENTS)
+    //  //broadcast the input file name to other nodes
+    //  MPI_Bcast(fname.c_str(),fname.size(),MPI_CHAR,0,OHMMS::Controller->getID());
+    //#endif
+
+    QMCMain* qmc    = 0;
+    bool validInput = false;
+    app_log() << "  Input file(s): ";
+    for (int k = 0; k < inputs.size(); ++k)
+      app_log() << inputs[k] << " ";
+    app_log() << std::endl;
+    qmc = new QMCMain(qmcComm);
+    if (inputs.size() > 1)
+      validInput = qmc->parse(inputs[qmcComm->getGroupID()]);
+    else
+      validInput = qmc->parse(inputs[0]);
+    if (validInput)
+      qmc->execute();
+
+    Libxml2Document timingDoc;
+    timingDoc.newDoc("resources");
+    output_hardware_info(qmcComm, timingDoc, timingDoc.getRoot());
+    TimerManager.output_timing(qmcComm, timingDoc, timingDoc.getRoot());
+    qmc->ptclPool->output_particleset_info(timingDoc, timingDoc.getRoot());
     if (OHMMS::Controller->rank() == 0)
     {
-      std::cerr << "No input file is given." << std::endl;
-      std::cerr << "Usage: qmcpack <input-files> " << std::endl;
+      timingDoc.dump(qmc->getTitle() + ".info.xml");
     }
+    TimerManager.print(qmcComm);
+    if (qmc)
+      delete qmc;
+    if (useGPU)
+      Finalize_CUDA();
     OHMMS::Controller->finalize();
-    return 1;
-  }
-  if (useGPU)
-    Init_CUDA();
-  //safe to move on
-  Communicate* qmcComm = OHMMS::Controller;
-  if (inputs.size() > 1)
-  {
-    qmcComm               = new Communicate(*OHMMS::Controller, inputs.size());
-    qmc_common.mpi_groups = inputs.size();
-  }
-  std::stringstream logname;
-  int inpnum          = (inputs.size() > 1) ? qmcComm->getGroupID() : 0;
-  std::string myinput = inputs[qmcComm->getGroupID()];
-  myinput             = myinput.substr(0, myinput.size() - 4);
-  logname << myinput;
-
-  if (qmcComm->rank() != 0)
-  {
-    outputManager.shutOff();
-    // might need to redirect debug stream to a file per rank if debugging is enabled
   }
-  if (inputs.size() > 1 && qmcComm->rank() == 0)
+  catch (const std::exception& e)
```

Comment:
Does std::bad_alloc give a reasonable message in what(), or should there be a separate catch for the out-of-memory condition?

Reply:
Well what() is all you have to call.  What is in it is implementation defined I believe.